doc/man7/passphrase-encoding.pod

```diff
@@ -0,0 +1,182 @@
+
+This section assumes that you know what pass phrase you should type in is, but
+that it may have been used in a different character encoding than your current
+environment gives you by default.
+For example, the pass phrase may have been used at a time when your default
+encoding was ISO-8859-1 (i.e. "naïve" resulting in the byte sequence 0x6E 0x61
+0xEF 0x76 0x65), and you're now in an environment where your default encoding
+is UTF-8 (i.e. "naïve" resulting in the byte sequence 0x6E 0x61 0xC3 0xAF 0x76
+0x65).
+Whenever it's mentioned that you should use a certain character encoding, it
+should be understood that you either change your environment to use the
+mentioned encoding by default when you type in your pass phrase, or use some
+suitable tool to convert your pass phrase from your default encoding to the
+target encoding.
+
+Also note that the sub-sections below discuss human readable pass phrases.
+This is particularly relevant for PKCS#12 objects, where human readable pass
+phrases are assumed.
+For other objects, it's as legitimate to use any byte sequence (such as a
+sequence of bytes from `/dev/urandom` that's been saved away), which makes any
+character encoding discussion irrelevant; in such cases, simply use the same
+byte sequence as it is.
+
+=head2 Creating new objects
+
+For creating new pass phrase protected objects, make sure the pass phrase is
+encoded using UTF-8.
+This is default on most modern Unixes, but may involve an effort on other
+platforms.
+Specifically for Windows, setting the environment variable
+C<OPENSSL_WIN32_UTF8> will have anything prompted on standard input converted
+to UTF-8.
+
+=head2 Opening existing objects
+
+For opening pass phrase protected objects where you know what character
+encoding was used for the encryption pass phrase, make sure to use the same
+encoding again.
+
+For opening pass phrase protected objects where the character encoding that was
+used is unknown, or where the producing application is unknown, try one of the
+following:
+
+=over 4
+
+=item 1.
+
+Try the password that you have as it is in the character encoding of your
+environment.
+It's possible that its byte sequence is exactly right.
+
+=item 2.
+
+Convert the pass phrase to UTF-8 and try with the result.
+Specifically with PKCS#12, this should open up any object that was created
+according to the specification.
+
+=item 3.
+
+Do a naïve (i.e. purely mathematical) ISO-8859-1 to UTF-8 conversion and try
+with the result.
+This differs from the previous attempt because ISO-8859-1 maps directly to
+U+0000 to U+00FF, which other non-UTF-8 character sets do not.
+
+This also takes care of the case when a UTF-8 encoded string was used with
+OpenSSL older than 1.1.0.
+(for example, C<ï>, which is 0xC3 0xAF when encoded in UTF-8, would become 0xC3
+0x83 0xC2 0xAF when re-encoded in the naïve manner.
+The PKCS#12 functions would then re-encode it back to 0xC3 0xA4, and make the
+BMSString 0x00 0xC3 0x00 0xA4 0x00 0x00)
```

Comment:
'BMPString', not 'BMSString'. And wording about what happens is misleading. PKCS#12 functions does not re-encode it back to 0xC3 0xA4 and then makes 0x00 0xC3 0x00 0xA4. Decode actually yields 0x00 0xC3 0x00 0xA4, there is no intermediate step with 0xC3 0xA4. "Then conversion to BMPString would yield 0x00 0xC3 0x00 0xA4, the erroneous/non-compliant encoding used by earlier OpenSSL versions." or something like that?

Reply:
Okie